components/script/dom/document.rs

```diff
@@ -5593,9 +5591,15 @@ impl DocumentMethods for Document {
     }
 
     // https://drafts.csswg.org/css-font-loading/#font-face-source
-    fn Fonts(&self, can_gc: CanGc) -> DomRoot<FontFaceSet> {
-        self.fonts
-            .or_init(|| FontFaceSet::new(&self.global(), None, can_gc))
+    fn Fonts(&self) -> DomRoot<FontFaceSet> {
+        match *self.fonts_state.borrow() {
+            FontsState::Unfulfilled(is_ready) => {
+                let fonts = FontFaceSet::new(&self.global(), None, is_ready);
+                *self.fonts_state.borrow_mut() = FontsState::Fulfilled(Dom::from_ref(&fonts));
```

Comment:
I suspect this will panic, since we have an outstanding borrow from the match statement.